File: app/models/booking.rb
class Booking < ActiveRecord::Base
  attr_accessible :hotel_id, :trip_id
  belongs_to :hotel
  belongs_to :trip

  def number_of_nights
    ( trip.number_of_days - 1 )
  end

  def cost_per_night
    hotel.price
  end
end

File: app/assets/javascripts/models/booking.js
EmberDay.Booking = DS.Model.extend({
  hotel:    DS.belongsTo('EmberDay.Hotel'),
  trip:     DS.belongsTo('EmberDay.Trip'),
  costPerNight:   DS.attr('number'),
  numberOfNights:   DS.attr('number'),

  totalCost: function() {
    return this.get('numberOfNights') * this.get('costPerNight');
  }.property('numberOfNights', 'costPerNight'),
});
